kore/src/Kore/Simplify/Exists.hs

```diff
@@ -70,6 +70,9 @@ import Kore.Rewrite.RewritingVariable (
 import qualified Kore.Simplify.AndPredicates as And (
     simplifyEvaluatedMultiPredicate,
  )
+import Kore.Simplify.Pattern (
+    simplifyPatternId,
```

Comment:
I think it's best if we move this into `MonadSimplify` as well. This way, we have access to it throughout the simplification code and we won't introduce new module dependencies.